### PR TITLE
Check machine status and log details if it is not running

### PR DESCRIPTION
**Pull request by ventifus**
### Which issue this PR addresses:

Provides more data for ARO-4247

### What this PR does / why we need it:

It is hard to troubleshoot worker node creation failures for new cluster installation. We wait for worker nodes, but sometimes they never show up because there was an Azure failure creating the VM. CAPI logs for machine creation happen too early and don't make it to Kusto.

### Test plan for issue:

Deploy test clusters and make sure we get the logs desired. Centraluseuap would be a good place to start if the error creating availability sets reproduces.

Happy-path logs look like the following:
```
2024-10-07T17:48:32.6932154Z time="2024-10-07T17:48:32Z" level=info msg="running step [Condition pkg/cluster.(*manager).minimumWorkerNodesReady, timeout 30m0s]" func="steps.Run()" file="pkg/util/steps/runner.go:56"
2024-10-07T17:48:32.7492655Z time="2024-10-07T17:48:32Z" level=info msg="Machine v4-e2e-v105110278-eas-lj5v8-worker-eastus1-gsz9j is Running; status: {\"conditions\":[{\"lastTransitionTime\":\"2024-10-07T17:22:58Z\",\"message\":\"machine successfully created\",\"reason\":\"MachineCreationSucceeded\",\"status\":\"True\",\"type\":\"MachineCreated\"}],\"metadata\":{},\"vmId\":\"/subscriptions/***/resourceGroups/aro-v4-e2e-v105110278-eastus/providers/Microsoft.Compute/virtualMachines/v4-e2e-v105110278-eas-lj5v8-worker-eastus1-gsz9j\",\"vmState\":\"Running\"}" func="cluster.(*manager).minimumWorkerNodesReady()" file="pkg/cluster/condition.go:41"
2024-10-07T17:48:32.7498981Z time="2024-10-07T17:48:32Z" level=info msg="Machine v4-e2e-v105110278-eas-lj5v8-worker-eastus2-kfkb7 is Running; status: {\"conditions\":[{\"lastTransitionTime\":\"2024-10-07T17:23:20Z\",\"message\":\"machine successfully created\",\"reason\":\"MachineCreationSucceeded\",\"status\":\"True\",\"type\":\"MachineCreated\"}],\"metadata\":{},\"vmId\":\"/subscriptions/***/resourceGroups/aro-v4-e2e-v105110278-eastus/providers/Microsoft.Compute/virtualMachines/v4-e2e-v105110278-eas-lj5v8-worker-eastus2-kfkb7\",\"vmState\":\"Running\"}" func="cluster.(*manager).minimumWorkerNodesReady()" file="pkg/cluster/condition.go:41"
2024-10-07T17:48:32.7504072Z time="2024-10-07T17:48:32Z" level=info msg="Machine v4-e2e-v105110278-eas-lj5v8-worker-eastus3-zdpfd is Running; status: {\"conditions\":[{\"lastTransitionTime\":\"2024-10-07T17:23:09Z\",\"message\":\"machine successfully created\",\"reason\":\"MachineCreationSucceeded\",\"status\":\"True\",\"type\":\"MachineCreated\"}],\"metadata\":{},\"vmId\":\"/subscriptions/***/resourceGroups/aro-v4-e2e-v105110278-eastus/providers/Microsoft.Compute/virtualMachines/v4-e2e-v105110278-eas-lj5v8-worker-eastus3-zdpfd\",\"vmState\":\"Running\"}" func="cluster.(*manager).minimumWorkerNodesReady()" file="pkg/cluster/condition.go:41"
2024-10-07T17:48:32.8130248Z time="2024-10-07T17:48:32Z" level=info msg="Node v4-e2e-v105110278-eas-lj5v8-worker-eastus1-gsz9j status: [{MemoryPressure False 2024-10-07 17:46:14 +0000 UTC 2024-10-07 17:41:07 +0000 UTC KubeletHasSufficientMemory kubelet has sufficient memory available} {DiskPressure False 2024-10-07 17:46:14 +0000 UTC 2024-10-07 17:41:07 +0000 UTC KubeletHasNoDiskPressure kubelet has no disk pressure} {PIDPressure False 2024-10-07 17:46:14 +0000 UTC 2024-10-07 17:41:07 +0000 UTC KubeletHasSufficientPID kubelet has sufficient PID available} {Ready True 2024-10-07 17:46:14 +0000 UTC 2024-10-07 17:41:08 +0000 UTC KubeletReady kubelet is posting ready status}]" func="cluster.(*manager).minimumWorkerNodesReady()" file="pkg/cluster/condition.go:61"
2024-10-07T17:48:32.8136352Z time="2024-10-07T17:48:32Z" level=info msg="Node v4-e2e-v105110278-eas-lj5v8-worker-eastus2-kfkb7 status: [{MemoryPressure False 2024-10-07 17:45:47 +0000 UTC 2024-10-07 17:44:15 +0000 UTC KubeletHasSufficientMemory kubelet has sufficient memory available} {DiskPressure False 2024-10-07 17:45:47 +0000 UTC 2024-10-07 17:44:15 +0000 UTC KubeletHasNoDiskPressure kubelet has no disk pressure} {PIDPressure False 2024-10-07 17:45:47 +0000 UTC 2024-10-07 17:44:15 +0000 UTC KubeletHasSufficientPID kubelet has sufficient PID available} {Ready True 2024-10-07 17:45:47 +0000 UTC 2024-10-07 17:44:15 +0000 UTC KubeletReady kubelet is posting ready status}]" func="cluster.(*manager).minimumWorkerNodesReady()" file="pkg/cluster/condition.go:61"
2024-10-07T17:48:32.8141057Z time="2024-10-07T17:48:32Z" level=info msg="Node v4-e2e-v105110278-eas-lj5v8-worker-eastus3-zdpfd status: [{MemoryPressure False 2024-10-07 17:47:34 +0000 UTC 2024-10-07 17:47:34 +0000 UTC KubeletHasSufficientMemory kubelet has sufficient memory available} {DiskPressure False 2024-10-07 17:47:34 +0000 UTC 2024-10-07 17:47:34 +0000 UTC KubeletHasNoDiskPressure kubelet has no disk pressure} {PIDPressure False 2024-10-07 17:47:34 +0000 UTC 2024-10-07 17:47:34 +0000 UTC KubeletHasSufficientPID kubelet has sufficient PID available} {Ready True 2024-10-07 17:47:34 +0000 UTC 2024-10-07 17:47:34 +0000 UTC KubeletReady kubelet is posting ready status}]" func="cluster.(*manager).minimumWorkerNodesReady()" file="pkg/cluster/condition.go:61"
2024-10-07T17:48:32.8144826Z time="2024-10-07T17:48:32Z" level=info msg="3 nodes ready" func="cluster.(*manager).minimumWorkerNodesReady()" file="pkg/cluster/condition.go:69"
```

### CODE REVIEW

**Review by tsatam — APPROVED** [review]
LGTM (after fixing the pointer dereference in the log statement) - I think this functionality could do with some more robust unit tests but that can be a follow-up effort.

**Review by shubhadapaithankar — DISMISSED** [review]

File: pkg/cluster/condition.go [hunk 1]
```diff
@@ -28,22 +31,45 @@ func (m *manager) apiServersReady(ctx context.Context) (bool, error) {
...
+	})
+	if err != nil {
+		m.log.Error(err)
+		return false, nil
+	}
+
+	readyWorkerMachines := 0
+	for _, machine := range machines.Items {
+		m.log.Infof("Machine %s is %s; status: %s", machine.Name, *machine.Status.Phase, string(machine.Status.ProviderStatus.Raw))
+		if *machine.Status.Phase == phaseRunning {
```
Comment (kimorris27) [inline pkg/cluster/condition.go:45]: Nit: Let's check for nil pointers here just to be super safe.

**Review by kimorris27 — CHANGES_REQUESTED** [review]
LGTM except for one spot where I think it's worth checking for nil pointers.

**Review by kimorris27 — APPROVED** [review]

**Review by cadenmarchese — APPROVED** [review]
lgtm

**Pull request merged**